runtimeconfig/manager_test.go

```diff
@@ -76,27 +76,133 @@ func newTestOverridesManagerConfig(t *testing.T, i int32) (*atomic.Int32, Config
 	}
 }
 
+func generateRuntimeFiles(t *testing.T, overrideStrings []string) ([]*os.File, error) {
+	var overrideFiles []*os.File
+
+	for count, override := range overrideStrings {
+		pattern := fmt.Sprintf("overrides-file-%d", count)
+		tempFile, err := os.CreateTemp("", pattern)
+		if err != nil {
+			return nil, err
+		}
+		_, err = tempFile.WriteString(override)
+		if err != nil {
+			return nil, err
+		}
+		overrideFiles = append(overrideFiles, tempFile)
+	}
+	t.Cleanup(func() {
+		require.NoError(t, cleanupOverridesFiles(overrideFiles))
+	})
```

Comment:
this would also clean up in case of partial success
```suggestion
	t.Cleanup(func() {
		require.NoError(t, cleanupOverridesFiles(overrideFiles))
	})
	for count, override := range overrideStrings {
		pattern := fmt.Sprintf("overrides-file-%d", count)
		tempFile, err := os.CreateTemp("", pattern)
		if err != nil {
			return nil, err
		}
		_, err = tempFile.WriteString(override)
		if err != nil {
			return nil, err
		}
		overrideFiles = append(overrideFiles, tempFile)
	}
```